cpp/include/cugraph/algorithms.hpp

```diff
@@ -24,12 +24,137 @@
 #include <cugraph/internals.hpp>
 #include <cugraph/legacy/graph.hpp>
 
+#include <cugraph-ops/aggregator.hpp>
 #include <cugraph-ops/graph/sampling.hpp>
 
 #include <raft/handle.hpp>
 
 namespace cugraph {
 
+/**
+ * @brief Computes the forward pass for aggregation layer
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ *
+ * @param handle                     RAFT handle object to encapsulate resources (e.g. CUDA stream,
+ * communicator, and handles to various CUDA libraries) to run graph algorithms.
+ * @param[in] message_flow_graph     input message flow graph.
+ * @param[in] dimension              dimensionality of the embeddings
+ * @param[in] leading_dimension      leading dimension for `aggregated_embeddings` array.
+ *                                   It must be at least `dimension`.
+ * @param[in] ignore_self            ignore the self node in the aggregation
+ * @param[in] op                     aggregation operation
+ * @param[in] input_embeddings       the input embeddings. [on device]
+ *                                   [dim = `*mfg.n_in_nodes x dim`]
+ * @param[out] aggregated_embeddings the output aggregated embeddings. [on device]
+ *                                   [dim = `mfg.n_out_nodes x ld`]
+ * @param[out] embedding_index       (optional) location of the max/min embeddings. This will be
+ *                                   written to only for kMin/kMax aggregations and
+ *                                   when this pointer is NOT a `nullptr`. [on device]
```

Comment:
>when this pointer is NOT a `nullptr`
=> Just to double check, this variable is std::optional, and are you saying we need to check both `embedding_index.has_value() && *(embedding_index) != nullptr`?